fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/AgilityContext.java

```diff
@@ -452,8 +452,8 @@ public <R> CompletableFuture<R> apply(Function<FDBRecordContext, CompletableFutu
 
         @Override
         public <R> CompletableFuture<R> applyInRecoveryPath(Function<FDBRecordContext, CompletableFuture<R>> function) {
-            // No recovery for a single user transaction
-            return CompletableFuture.completedFuture(null);
+            // Best effort - skip ensureOpen, ignore exceptions.
+            return function.apply(callerContext).exceptionally(ex -> null);
```

Comment:
I'm assuming this change was found necessary by the additional coverage in the test?

Reply:
Yes. Without it the non-agile version had failed.